iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_receiver.hpp

```diff
@@ -34,6 +34,23 @@ enum class ChunkReceiveResult
     NO_CHUNK_AVAILABLE
 };
 
+/// @brief Converts the ChunkReceiveResult to a string literal
+/// @param[in] value to convert to a string literal
+/// @return pointer to a string literal
+inline constexpr const char* asStringLiteral(const ChunkReceiveResult value) noexcept;
+
+/// @brief Convenience stream operator to easily use the `asStringLiteral` function with std::ostream
+/// @param[in] stream sink to write the message to
+/// @param[in] value to convert to a string literal
+/// @return the reference to `stream` which was provided as input parameter
+inline std::ostream& operator<<(std::ostream& stream, ChunkReceiveResult value) noexcept;
```

Comment:
I would propose a template here which is explicitly instantiated in the cpp file.
```cpp
// in header file
template<typename T>
T& operator<<(T& stream, ChunkReceiveResult value) noexcept;

// in cpp file
T& operator<<(T& stream, ChunkReceiveResult value) noexcept
{
    stream << asStringLiteral(value);
    return stream;
}

template log::Stream& operator<<(log::LogStream & stream, ChunkReceiveResult value) noexcept;
template std::ostream& operator<<(std::ostream & stream, ChunkReceiveResult value) noexcept;
```

Or we could also keep this inline in the `inl` file. But why have code duplication and restrict a generic approach at the same time?

Reply:
Do we need to instantiate it?
In this case there is already an `inl` available but no `cpp`.

Since I used this pattern already at multiple locations, is it okay for you that I adjust the new code here and create an issue to also change it at the other locations when my next PR is ready for review?

Reply:
@elfenpiff I have another idea. What do you think about having something like this in the hoofs
```cpp
template <typename T>
constexpr const char* asStringLiteral(const T value)
{
    static_assert(always_false_v(T), "'asStringLiteral' is not specialized for the specified type");
}
```
The logger could then do a blanket implementation like
```
template <typename T>
LogStream& operator<<(LogStream& stream, T value) noexcept
{
    stream << asStringLiteral(value);
    return stream;
}
```

Unfortunately we cannot do the same for `std::ostream` but most probably we also do not want to have `ostream` in the certified codebase. Therefore it would make sense to have all the `std::ostream& operator<<(std::ostream& stream, Foo value) noexcept` in a separate hpp/cpp file which will be in the `aux` namespace and empty for the certified subset.

If you agree on this, I would keep the current PR as is and implement the proposal while waiting for review on my next PR.

Reply:
I like it!

Reply:
Created #1066